app/code/community/Nexcessnet/Turpentine/Model/Observer/Esi.php

```diff
@@ -528,45 +528,29 @@ protected function _checkIsEsiUrl($url) {
         return ! $this->_checkIsNotEsiUrl($url);
     }
 
-    public function hookToControllerActionPreDispatch($observer) {
```

Comment:
This method was unnecessary, as its only responsibility is routing to another method in the same class. With the changes to observe a more specific event name, I made it route directly to the respective methods doing actual work and eliminated this.